File: backend/app/api/v1/pipeline.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.auth import get_current_user
from app.services.rss_fetcher import fetch_all_sources
from app.services.news_processor import process_pending_news
from app.services.similarity_checker import check_similarity_batch
from app.services.importance_judge import judge_importance_batch
from app.services.pipeline_orchestrator import run_full_pipeline

router = APIRouter(prefix="/pipeline", tags=["pipeline"], dependencies=[Depends(get_current_user)])


@router.post("/fetch-rss")
async def trigger_fetch_rss(db: AsyncSession = Depends(get_db)):
    return await fetch_all_sources(db)


@router.post("/process-news")
async def trigger_process_news(db: AsyncSession = Depends(get_db)):
    return await process_pending_news(db)


@router.post("/check-similarity")
async def trigger_check_similarity(db: AsyncSession = Depends(get_db)):
    return await check_similarity_batch(db)


@router.post("/judge-importance")
async def trigger_judge_importance(db: AsyncSession = Depends(get_db)):
    return await judge_importance_batch(db)


@router.post("/run-full")
async def trigger_full_pipeline(db: AsyncSession = Depends(get_db)):
    return await run_full_pipeline(db)

File: frontend/src/api/pipeline.ts
import client from './client'

export async function triggerFetchRss() {
  const resp = await client.post('/pipeline/fetch-rss')
  return resp.data
}

export async function triggerProcessNews() {
  const resp = await client.post('/pipeline/process-news')
  return resp.data
}

export async function triggerCheckSimilarity() {
  const resp = await client.post('/pipeline/check-similarity')
  return resp.data
}

export async function triggerJudgeImportance() {
  const resp = await client.post('/pipeline/judge-importance')
  return resp.data
}

export async function triggerFullPipeline() {
  const resp = await client.post('/pipeline/run-full')
  return resp.data
}
